iceoryx_binding_c/test/moduletests/test_subscriber.cpp

```diff
@@ -245,3 +259,52 @@ TEST_F(iox_sub_test, sendingTooMuchLeadsToLostChunks)
 
     EXPECT_TRUE(iox_sub_has_lost_chunks(m_sut));
 }
+
+TEST_F(iox_sub_test, attachingToWaitSetWorks)
+{
+    EXPECT_EQ(iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 0, NULL), WaitSetResult_SUCCESS);
+    EXPECT_EQ(m_waitSet->size(), 1);
+}
+
+TEST_F(iox_sub_test, detachingFromWaitSetWorks)
+{
+    iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 0, NULL);
+    iox_sub_detach_ws(m_sut);
+    EXPECT_EQ(m_waitSet->size(), 0);
+}
+
+TEST_F(iox_sub_test, hasNewSamplesTriggersWaitSetWithCorrectTriggerId)
+{
+    iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 587, NULL);
+
+    this->Subscribe(&m_portPtr);
+    m_chunkPusher.tryPush(m_memoryManager.getChunk(100));
+
+    auto triggerVector = m_waitSet->wait();
+
+    ASSERT_EQ(triggerVector.size(), 1);
+    EXPECT_EQ(triggerVector[0].getTriggerId(), 587);
+}
+
+TEST_F(iox_sub_test, hasNewSamplesTriggersWaitSetWithCorrectCallback)
+{
+    iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 0, iox_sub_test::triggerCallback);
+
+    this->Subscribe(&m_portPtr);
+    m_chunkPusher.tryPush(m_memoryManager.getChunk(100));
+
+    auto triggerVector = m_waitSet->wait();
+
+    ASSERT_EQ(triggerVector.size(), 1);
+    triggerVector[0]();
```

Comment:
So what happens if you call multiple times `triggerVector[0]();`. is this allowed and should be tested? Or is this some kind of misuse?

Reply:
It is allowed and no misuse. Test is added in trigger state test since it is trigger state functionality. Here I test only if the callback is correctly attached by the subscriber.